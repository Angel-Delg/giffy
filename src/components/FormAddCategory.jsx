import PropTypes from 'prop-types'
import { useState } from 'react'

export function FormAddCategory ( { onNewValue } ){

    const [category, setCategory] = useState('')

    const handleChangeCategory = ( { target } ) => {
        setCategory( target.value.trim() )
    }
    
    // handleCategory
    const handleSubmitCategory = event => {
        event.preventDefault()

        if ( category.length === 0 ) {
            return
        }

        // Si no se cumple agregala
        onNewValue( category )
        setCategory('')
    }

    return(
        <form 
            className='containerForm' 
            onSubmit={ handleSubmitCategory }>

            
            <h1 className="titleApp">GiffyExpertApp</h1>

            <input 
                onChange={ handleChangeCategory } 
                value={ category } 
                type="text" 
                placeholder='Search giffy' 
                required  
            />
            <button 
                className='ButtonCategory'
            > 
             search
            </button>
        </form>
    )
}

FormAddCategory.propTypes = {
    onNewValue: PropTypes.func.isRequired
}