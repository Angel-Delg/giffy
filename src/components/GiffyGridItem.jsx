import PropTypes from 'prop-types'

export const GiffyGridItem = ({title, urlGiffy }) => {
  return (
    <section className="card">
      <img src={urlGiffy} alt={title} />
      <p data-testId="test-title">{ title }</p>
    </section>
  )
}

GiffyGridItem.propTypes = {
  title: PropTypes.string.isRequired,
  urlGiffy: PropTypes.string.isRequired
}