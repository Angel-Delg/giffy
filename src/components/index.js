export * from "/FormAddCategory"
export * from "/GiffyGrid"
export * from "/GiffyGridItem"