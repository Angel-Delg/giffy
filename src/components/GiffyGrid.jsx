import useFetchGifs from '../hooks/useFetchGifs'
import { GiffyGridItem } from './GiffyGridItem'

// Cargamos el spinner component
import { Loading } from './Loading'

const GiffyGrid = ({ category }) => {

    const { giffys, isLoading } = useFetchGifs( category  )

    return (
        <>
            <h2>{ category }</h2>
            { isLoading && ( <Loading />) }
            <article className='card-grid'>
                {
                    giffys.map( gif => (
                        <GiffyGridItem
                            key={ gif.id }
                            {...gif }
                        />
                    ))
                }
            </article>
        </>
    )
}

export default GiffyGrid