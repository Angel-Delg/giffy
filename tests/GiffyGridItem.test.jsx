import { render, screen } from "@testing-library/react"
import { GiffyGridItem } from "../src/components/GiffyGridItem"

describe('<GiffyGridItem />', () => {
    const title = 'saitama'
    const URL = 'https://unsplash/saitama.jpg'

    test('This is test GiffyGridItem.jsx',() => {

        const { container, getByText } = render( <GiffyGridItem title={ title } urlGiffy={ URL }/> )

        // snapshot toma una fotografia
        expect( container ).toMatchSnapshot();
    })

    test('Title toBeTruthy in the component', () => {

        const { container, getByText, getByTestId } = render( <GiffyGridItem title={ title } urlGiffy={ URL }/> )
        // Obtenemos el texto y lo que espero es que se encuentre el texto
        expect( getByText( title )).toBeTruthy()
        expect( getByTestId('test-title').innerHTML ).toBe( title )
    })

    test('Debe mostrar la imagen con el url indicado', () => {

        render(<GiffyGridItem title={title} urlGiffy={URL} />)
        screen.debug()

        const { src, alt } = screen.getByRole('img')
        expect( src ).toBe(URL)
        expect( alt ).toBe(title)

    }) 

    test('Debe mostrar el titulo en el componente', () => {
        const { container } = render(<GiffyGridItem title={title} urlGiffy={URL} />)
        expect(container.querySelector('p').innerHTML).toBe(title)
    })

})