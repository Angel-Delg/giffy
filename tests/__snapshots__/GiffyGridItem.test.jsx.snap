// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GiffyGridItem /> This is test GiffyGridItem.jsx 1`] = `
<div>
  <section
    class="card"
  >
    <img
      alt="saitama"
      src="https://unsplash/saitama.jpg"
    />
    <p
      data-testid="test-title"
    >
      saitama
    </p>
  </section>
</div>
`;
